packages/asc-ui/src/components/Button/ButtonStyle.ts

```diff
@@ -1,41 +1,177 @@
-      : getColor('primary')({ theme })};
-  ${({ theme, color }) =>
-    !color &&
-    css`
-      border: 1px solid ${getColor('primary')({ theme })};
-    `}
-
-  &:hover {
-    ${({ theme, color }) =>
-      !color &&
-      css`
+const getVariant = () => ({
+  theme,
+  variant,
+  color,
+}: {
+  theme: any
+  variant?: keyof typeof ButtonVariants
+  color?: Theme.TypeLevel
+}) => {
+  switch (variant) {
+    case 'primary':
+      return css`
+        background-color: ${themeColor('primary')({ theme })};
+        color: ${readableColor(themeColor('primary')({ theme }))};
+        ${svgFill('tint', 'level1')};
+
+        &:hover {
+          background-color: ${darken(0.1, themeColor('primary')({ theme }))};
+        }
+      `
+
+    case 'secondary':
+      return css`
+        background-color: ${themeColor('secondary')({ theme })};
+        color: ${themeColor('tint', 'level1')({ theme })};
+        ${svgFill('tint', 'level1')};
+
+        &:hover {
+          background-color: ${darken(0.1, themeColor('secondary')({ theme }))};
+        }
+      `
+
+    case 'tertiary':
+      return css`
+        background-color: ${themeColor('tint', 'level4')({ theme })};
+
+        &:hover {
+          background-color: ${darken(
+            0.1,
+            themeColor('tint', 'level4')({ theme }),
+          )};
+        }
+      `
+
+    case 'primaryInverted':
+      return css`
+        color: ${getColor('primary')({ theme })};
         outline: 1px solid ${getColor('primary')({ theme })};
-      `}
-  }
+        ${svgFill('primary')};
+
+        &:hover {
+          outline-width: 2px;
+        }
+      `
+
+    case 'blank':
+      return css`
+        ${svgFill('tint', 'level7')}
+        &:hover {
+          background-color: ${themeColor('tint', 'level3')({ theme })};
+        }
+      `
+    default:
+      // About-to-be-deprecated codeblock to support the 'color' prop
+      return css`
+        color: ${color
+          ? readableColor(getColor(color)({ theme }))
+          : getColor('primary')({ theme })};
+        ${color &&
+          css`
+            background: ${themeColor(color)({ theme })};
+          `}
 
-  & > ${/* sc-selector */ IconStyle}:first-child {
-    margin-right: 5px;
+        ${!color &&
+          css`
+            outline: 1px solid ${getColor('primary')({ theme })};
+          `}
+
+        &:hover {
+          background: ${color
+            ? themeColor(color, 'dark')({ theme })
+            : themeColor('tint', 'level3')({ theme })};
+          ${!color &&
+            css`
+              outline-width: 2px;
+            `}
+        }
+      `
   }
+}
+
+export type Props = {
+  /**
+   * Deprecated: Use variant instead. Pass the theme-color.
+   * @deprecated
+   */
+  color?: Theme.TypeLevel
+  /**
+   * Deprecated: use size to create a button with equal width and height
+   * @deprecated
+   */
+  square?: boolean
+  /**
+   * Set an equal height and width
+   */
+  size?: number
```

Comment:
How can you set an equal height and width with this?

Reply:
By using the [size](https://github.com/Amsterdam/amsterdam-styled-components/pull/91/files#diff-317386189eb84716077d89eb245d5142R150) util from polished. Will rewrite this to height: size, width: size when we decide to really get rid of polished